File: BI-TJV/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/table/TableCreateListener.java
package cz.wrzecond.restsys.table;

import android.app.Dialog;
import android.content.DialogInterface;
import android.widget.ArrayAdapter;
import android.widget.Button;

import androidx.appcompat.widget.AppCompatSpinner;

import cz.wrzecond.restsys.R;
import cz.wrzecond.restsys.CreateListener;
import wrzecond.table.TjvTableCreateDTO;
import wrzecond.table.TjvTableReadDTO;
import wrzecond.table.TjvTableType;

public class TableCreateListener extends CreateListener<TjvTableReadDTO> {

    public TableCreateListener (TableActivity activity, Dialog dialog, TjvTableReadDTO dto) { super(activity, dialog, dto); }

    @Override
    public void onShow (DialogInterface di) {
        // Init dialog
        AppCompatSpinner tableType = dialog.findViewById(R.id.create_table_type);
        ArrayAdapter<TjvTableType> types = new ArrayAdapter<>(activity, R.layout.support_simple_spinner_dropdown_item);
        types.addAll(TjvTableType.values());
        tableType.setAdapter(types);

        if ( dto != null ) {
            tableType.setSelection(dto.getType().ordinal());

            Button button = dialog.findViewById(R.id.create_button);
            button.setText(R.string.button_edit);
        }

        // Perform
        dialog.findViewById(R.id.create_button).setOnClickListener( view -> {
            // Create DTO
            TjvTableCreateDTO dto = new TjvTableCreateDTO((TjvTableType) tableType.getSelectedItem());
            scheduleRequest(dto, "/tables/");
        });
    }

}

File: BI-KOT/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/table/TableCreateListener.kt
package cz.wrzecond.restsys.table

import android.app.Dialog
import android.content.DialogInterface
import android.widget.ArrayAdapter
import android.widget.Button
import androidx.appcompat.widget.AppCompatSpinner
import cz.wrzecond.restsys.CreateListener
import cz.wrzecond.restsys.R
import wrzecond.TjvTableReadDTO
import wrzecond.TjvTableType
import wrzecond.TjvTableUpdateDTO

class TableCreateListener (activity: TableActivity, dialog: Dialog, dto: TjvTableReadDTO?)
: CreateListener<TjvTableReadDTO>(activity, dialog, dto) {

    override fun onShow (di: DialogInterface) {
        // Set type spinner
        val types = ArrayAdapter<TjvTableType>(activity, R.layout.support_simple_spinner_dropdown_item)
        types.addAll(TjvTableType.values().toMutableList())

        val tableType = dialog.findViewById<AppCompatSpinner>(R.id.create_table_type)
        tableType.adapter = types

        // Init button
        val button = dialog.findViewById<Button>(R.id.create_button)

        // Existing type
        if (dto != null) {
            tableType.setSelection(dto.type.ordinal)
            button.setText(R.string.button_edit)
        }

        // Perform creation / edit
        button.setOnClickListener {
            val dto = TjvTableUpdateDTO(tableType.selectedItem as TjvTableType)
            scheduleRequest(dto, "/tables/")
        }
    }

}